components/brave_ads/core/internal/account/user_data/fixed/summary_user_data_util.cc

```diff
@@ -11,11 +11,8 @@ AdTypeBucketMap BuildBuckets(const PaymentTokenList& payment_tokens) {
   AdTypeBucketMap buckets;
 
   for (const auto& payment_token : payment_tokens) {
-    const std::string ad_type = ToString(payment_token.ad_type);
-    const std::string confirmation_type =
-        ToString(payment_token.confirmation_type);
-
-    buckets[ad_type][confirmation_type]++;
+    buckets[ToString(payment_token.ad_type)]
```

Comment:
I noticed that the definition of the map is:

```c++
using AdTypeBucketMap =
    base::flat_map</*ad_type=*/std::string, ConfirmationTypeBucketMap>;
```

I suspect this flat_map's key could be `AdType`. That would have better performance/memory-wise in general.

Reply:
To clarify, you can defer the conversion `ToString` all the way to the last operation in `BuildSummaryUserData`, constraining this flat_map in the process to a more concrete, and compact type in the process.

If there are cases like that, they would benefit of moving the enum as far as you can get the value strongly typed to.

Reply:
Cleaned up, thanks